### PR TITLE
Ensure Validator Has No Breaking Changes in New Slasher

### PR DESCRIPTION
**Pull request by rauljordan**
**What type of PR is this?**

> Other

**What does this PR do? Why is it needed?**

The optimized slasher in the feature branch #8340 has some breaking changes to the way validator clients do remote slashing protection. This is **unacceptable**. Instead, we keep the current behavior but add a new flag `--enable-new-external-slashing-protection` which will enable the functionality of the new slasher without breaking existing setups.

This PR adds back the files for remote slashing protection as it exists today in Prysm, which should not have been deleted in #8340.

**Which issues(s) does this PR fix?**

Part of #8331 

**Other notes for review**

### CODE REVIEW

File: validator/slashing-protection/external.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package slashingprotection
```
Comment (rauljordan) [inline validator/slashing-protection/external.go:1]: Nothing new: this already exists in `develop`

File: validator/slashing-protection/external_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
+package slashingprotection
```
Comment (rauljordan) [inline validator/slashing-protection/external_test.go:1]: Nothing new: this already exists in `develop`

File: validator/slashing-protection/iface/protector.go [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package iface
```
Comment (rauljordan) [inline validator/slashing-protection/iface/protector.go:1]: Nothing new: this already exists in `develop`

File: validator/slashing-protection/slasher_client.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
+package slashingprotection
```
Comment (rauljordan) [inline validator/slashing-protection/slasher_client.go:1]: Nothing new: this already exists in `develop`

File: validator/slashing-protection/slasher_client_test.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package slashingprotection
```
Comment (rauljordan) [inline validator/slashing-protection/slasher_client_test.go:1]: Nothing new: this already exists in `develop`

File: validator/testing/mock_protector.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
+package testing
```
Comment (rauljordan) [inline validator/testing/mock_protector.go:1]: Nothing new: this already exists in `develop`

File: cmd/validator/flags/flags.go [hunk 7]
```diff
@@ -61,7 +61,17 @@ var (
 		Usage: "RPC port exposed by a validator client",
 		Value: 7000,
 	}
-
+	// SlasherRPCProviderFlag defines a slasher node RPC endpoint.
```
Comment (rauljordan) [inline cmd/validator/flags/flags.go:64]: These already exist in develop but were deleted in feature/slasher

File: validator/client/slashing_protection_interchange_test.go [hunk 8]
```diff
@@ -1,168 +0,0 @@
-package client
```
Comment (terencechain) [inline validator/client/slashing_protection_interchange_test.go:1]: Why are we deleting this test?

Comment (rauljordan) [reply validator/client/slashing_protection_interchange_test.go]: nice catch, super important we do not lose this test

**Review by terencechain — APPROVED** [review]

**Pull request merged**